tck/src/main/java/org/eclipse/microprofile/lra/tck/TckContextTests.java

```diff
@@ -0,0 +1,201 @@
+        String status = invoke(STATUS_PATH, HttpMethod.GET, lra, 202, ContextTckResource.EndPhase.SUCCESS, 200);
+        assertEquals(testName.getMethodName() + ": participant is not completing", ParticipantStatus.Completing.name(), status);
+
+        // clear the EndPhase override data so that the next status request returns completed or compensated
+        invoke(STATUS_PATH, PUT, lra, 200, ContextTckResource.EndPhase.SUCCESS, 200);
+
+        // trigger a replay of the end phase
+        replayEndPhase(TCK_CONTEXT_RESOURCE_PATH);
+
+        // and verify that the resoure was asked to complete
+        status = invoke(STATUS_PATH, HttpMethod.GET, lra, 200, ContextTckResource.EndPhase.SUCCESS, 200);
+        assertEquals(testName.getMethodName() + ": participant is not completed", ParticipantStatus.Completed.name(), status);
+    }
+
+    @Test
+    public void testLeave() {
+        // call a resource that begins but does not ends an LRA
+        String lra = invoke(NEW_LRA_PATH, PUT, null, 200, ContextTckResource.EndPhase.SUCCESS, 200);
+        // verfiy that the resource is active
+        String status = invoke(STATUS_PATH, HttpMethod.GET, lra, 200, ContextTckResource.EndPhase.SUCCESS, 200);
+
+        assertEquals(testName.getMethodName() + ": participant is not active", ParticipantStatus.Active.name(), status);
+
+        // ask the resource to leave the active LRA
+        invoke(LEAVE_PATH, PUT, lra);
+
+        // end the LRA via a different resource since using the same one will re-enlist it
+        lraClient.closeLRA(lra);
+
+        // verify that the resource was not asked to complete
+        String completions = invoke(METRIC_PATH + "/" + Complete.class.getName(), GET, lra);
+
+
+        assertEquals(testName.getMethodName() + ": Resource left but was still asked to complete",
+                Integer.toString(0), completions);
+    }
+
+    @Test
+    public void testForget() {
+        // call a resource that begins and ends an LRA and coerces the resource to return FAILED when asked to complete
+        String lra = invoke(REQUIRED_LRA_PATH, PUT, null, 200, ContextTckResource.EndPhase.FAILED, 500);
+
+        // trigger a replay attempt
+        replayEndPhase(TCK_CONTEXT_RESOURCE_PATH);
+
+        // the implementation should have called status which will have returned 500
+        String count1 = invoke(METRIC_PATH + "/" + Status.class.getName(), GET, lra);
+        assertEquals(testName.getMethodName() + " resource status should have been called", "1", count1);
+
+        // the implementation should not call forget until it knows the particpant status
+        String count2 = invoke(METRIC_PATH + "/" + Forget.class.getName(), GET, lra);
+        assertEquals(testName.getMethodName() + " resource forget should not have been called", "0", count2);
+
+        // clear the fault
+        invoke(STATUS_PATH, PUT, lra, 200, ContextTckResource.EndPhase.FAILED, 200);
```

Comment:
There should be no confusion since the semantics of PUT and GET are well known in REST based systems:
- GET is for reading a resource representation
- PUT is for updating the resource representation

Reply:
@mmusgrov sure. I meant that it is easy to miss that this is actually a PUT invoke when it's placed on STATUS_PATH so I really just wanted to rename it to something like CLEAR_PATH or similar. But if you don't want to do this it's not big deal.

Reply:
But then it would not be obvious that the same resource is being accessed.

Reply:
I will add a commit that adds more documentation to clarify how the resource and tests interact ...

Reply:
@xstefank I added more comments